sky/services/oknet/src/org/domokit/oknet/UrlLoaderImpl.java

```diff
@@ -139,8 +143,37 @@ public void start(UrlRequest request, StartResponse callback) {
             return;
         }
 
+        RequestBody body = null;
+        if (request.body != null && request.body[0] != null) {
+          DataPipe.ConsumerHandle consumer = request.body[0];
+          ByteArrayOutputStream bodyStream = new ByteArrayOutputStream();
+          WritableByteChannel dest = Channels.newChannel(bodyStream);
+          do {
+              try {
+                  ByteBuffer buffer = consumer.beginReadData(0, DataPipe.ReadFlags.NONE);
+                  if (buffer.capacity() == 0) break;
+                  dest.write(buffer);
+                  consumer.endReadData(buffer.capacity());
+              } catch (IOException e) {
+                  throw new MojoException(e);
+              } catch (MojoException e) {
+                  // No one read the pipe, they just closed it.
+                  if (e.getMojoResult() == MojoResult.FAILED_PRECONDITION) {
+                      break;
+                  } else if (e.getMojoResult() == MojoResult.SHOULD_WAIT) {
+                      mCore.wait(consumer, Core.HandleSignals.READABLE, -1);
```

Comment:
This is a synchronous wait on this thread.  Rather than blocking this thread (which will block all input events), should we move this wait onto a background thread?